rails_docker/docker-compose.yml

```diff
@@ -1,36 +1,37 @@
-version: '3'
+version: '3.2'
+
 services:
   db:
-    image: postgres:9.6.1
+    image: postgres:10.1
     container_name: #{app_name}_db
     environment:
-      - POSTGRES_DB=#{app_name}_development
+      - POSTGRES_DB=#{app_name}_production
     ports:
-      - "5432:5432"
+      - "5432"
 
   web:
-    build: .
-    container_name: '#{app_name}_web'
-    command: bash -c "bundle check || bundle install && rm -f tmp/pids/server.pid && bundle exec rails s -p 3000 -b '0.0.0.0'"
+    build:
+      context: .
+      cache_from:
+        - ${DOCKER_IMAGE}
+    image: ${DOCKER_IMAGE}
+    container_name: #{app_name}_web
+    command: bin/start.sh
     ports:
-      - "3000:3000"
-    stdin_open: true
-    tty: true
-    volumes:
-      - '.:/#{app_name}'
-      - "ruby-bundle:/bundle"
-
-  test:
-    build: .
-    image: #{app_name}_test
-    entrypoint: bin/test.sh
+      - "${PORT}:${PORT}"
     stdin_open: true
     tty: true
+    depends_on:
+      - db
     volumes:
-      - .:/#{app_name}
       - "ruby-bundle:/bundle"
     environment:
```

Comment:
@olivierobert its very likely that we will have many env variables. 

Lets use [env_file](https://docs.docker.com/compose/environment-variables/#the-env_file-configuration-option) configuration option instead, so that we dont have to change `docker-compose.yml` file each time we add a new env

Reply:
I added the env file `.env` which is loaded automatically but then we would need to split it be environments e.g. `.env.test` and `.env.production`. My main motivation not to do it is that it's not supported in Heroku and it would be better to use docker-override yml file.